runners/samza/src/main/java/org/apache/beam/runners/samza/SamzaJobServerDriver.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.runners.samza;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.Nullable;
+import org.apache.beam.model.pipeline.v1.RunnerApi;
+import org.apache.beam.runners.core.construction.PipelineOptionsTranslation;
+import org.apache.beam.runners.fnexecution.GrpcFnServer;
+import org.apache.beam.runners.fnexecution.ServerFactory;
+import org.apache.beam.runners.fnexecution.artifact.BeamFileSystemArtifactStagingService;
+import org.apache.beam.runners.fnexecution.jobsubmission.InMemoryJobService;
+import org.apache.beam.runners.fnexecution.jobsubmission.JobInvocation;
+import org.apache.beam.runners.fnexecution.jobsubmission.JobInvoker;
+import org.apache.beam.vendor.protobuf.v3.com.google.protobuf.Struct;
+import org.kohsuke.args4j.CmdLineException;
+import org.kohsuke.args4j.CmdLineParser;
+import org.kohsuke.args4j.Option;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Driver program that starts a job server. */
+public class SamzaJobServerDriver {
+  private static final Logger LOG = LoggerFactory.getLogger(SamzaJobServerDriver.class);
+
+  private final ServerConfiguration config;
+
+  /** Configuration for the jobServer. */
+  private static class ServerConfiguration {
+    @Option(name = "--job-port", usage = "The job service port. (Default: 11440)")
+    private int jobPort = 11440;
+
+    @Option(name = "--control-port", usage = "The FnControl port. (Default: 11441)")
+    private int controlPort = 11441;
+  }
+
+  private SamzaJobServerDriver(ServerConfiguration config) {
+    this.config = config;
+  }
+
+  public static void main(String[] args) throws Exception {
+    ServerConfiguration configuration = new ServerConfiguration();
```

Comment:
I think this deserves a "final" here, and the line below too.

Reply:
done